File: app/src/main/java/com/meteorhead/leave/binding/adapters/ListViewBindingAdapters.java
package com.meteorhead.leave.binding.adapters;

import android.databinding.BindingAdapter;
import android.support.annotation.LayoutRes;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import com.meteorhead.leave.models.Leave;
import java.util.List;

/**
 * Created by wierzchanowskig on 19.02.2017.
 */

public class ListViewBindingAdapters {
    @BindingAdapter(value = { "items", "itemLayout" }, requireAll = false)
    public static void bindList(ListView listView, List<Leave> leaveList,
        @LayoutRes int itemLayoutId) {
        if(leaveList == null) {
            return;
        }
        if (itemLayoutId == 0) {
            itemLayoutId = android.R.layout.simple_list_item_1;
        }
        listView.setAdapter(new ArrayAdapter<>(listView.getContext(), itemLayoutId,
            android.R.id.text1, leaveList));
    }
}

File: app/src/main/java/com/meteorhead/leave/binding/adapters/ListViewBindingAdapters.kt
package com.meteorhead.leave.binding.adapters

import android.databinding.BindingAdapter
import android.support.annotation.LayoutRes
import android.widget.ArrayAdapter
import android.widget.ListView
import com.meteorhead.leave.data.Leave

/**
 * Created by wierzchanowskig on 19.02.2017.
 */

@BindingAdapter(value = *arrayOf("items", "itemLayout"), requireAll = false)
fun bindLeaveList(listView: ListView, leaveList: List<Leave>?,
             @LayoutRes itemLayoutId: Int) {
    var itemLayoutId = itemLayoutId
    if (leaveList == null) {
        return
    }
    if (itemLayoutId == 0) {
        itemLayoutId = android.R.layout.simple_list_item_1
    }
    listView.adapter = ArrayAdapter(listView.context, itemLayoutId,
                                    android.R.id.text1, leaveList)
}

